### ISSUE TITLE
jax.experimental.sparse.coo_fromdense gives incorrect second derivative when nse includes padded entries

### ISSUE THREAD

**Issue by ALinrunrun** [jax-ml/jax#40624]
### Description

`jax.experimental.sparse.coo_fromdense` appears to produce an incorrect second-order reverse-mode derivative when `nse` is larger than the number of true nonzero entries.

In the example below, the dense input has three true nonzero entries, while `nse=5`. The extra `out.data` entries are zero padding and should remain constant zero. Therefore, the local scalar function should be:

```
(t + 1)**2 + t**2 + (t**2 + 2)**2
```

At `t = 0.02`, the expected second derivative is `12.0048`. However, JAX returns `16.0048`.

### Minimal reproducible example

```
#!/usr/bin/env python3
import os
import sys
import math

os.environ.setdefault("CUDA_VISIBLE_DEVICES", "")
os.environ.setdefault("OMP_NUM_THREADS", "1")
os.environ.setdefault("JAX_PLATFORMS", "cpu")

try:
    import jax
    import jax.numpy as jnp
    import jax.experimental.sparse
except ImportError as e:
    print(f"missing dep: {e}")
    sys.exit(2)

jax.config.update("jax_enable_x64", True)

def target(t):
    z = jnp.zeros_like(t)
    a = jnp.stack((t + 1, z, -t, z, z, t * t + 2)).reshape((2, 3)).T
    out = jax.experimental.sparse.coo_fromdense(
        a,
        nse=5,
        index_dtype=jnp.int32,
    )
    return jnp.sum(out.data ** 2)

x = jnp.asarray(0.02, dtype=jnp.float64)
expected = 12.0048

actual = float(jax.grad(jax.grad(target))(x))

print("actual:", actual)
print("expected:", expected)

if not (math.isfinite(actual) and abs(actual - expected) <= 1e-12):
    sys.exit(0)
sys.exit(1)
```

### Actual result

```
actual: 16.0048
expected: 12.0048
```

### Expected result

```
12.0048
```

The expected value follows from differentiating the scalar function formed by the true nonzero entries:

```
(t + 1)**2 + t**2 + (t**2 + 2)**2
```

The padded entries introduced by `nse=5` should remain constant zero and should not contribute to the second derivative. JAX instead returns `16.0048`, which suggests that padded sparse entries may be leaking into the second-order autodiff result.

### System info (python version, jaxlib version, accelerator, etc.)

jax: 0.10.1
jaxlib: 0.10.1
numpy: 2.4.6
python: 3.13.13
platform: Ubuntu 24.04.3 LTS, Linux x86_64
accelerator: CPU backend

Note: GPU is present, but CUDA-enabled jaxlib is not installed, so JAX falls back to CPU.
